refactor: split into multiple files and added build-system

749 opened the pull request:
Closes #13 
Closes #5
The github build still needs debugging :-)

sybit-jgr committed:
refactor: split into multiple files and added build-system

shawly commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,42 @@
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - run: apt update && apt install unzip zip
```
```suggestion
      - run: apt-get update && apt-get install -y unzip zip
```

shawly replied:
Looking at the current workflow it seems that we don't even need to run this since unzip and zip are already packaged with the ubuntu runner anyways

shawly commented on package.json:
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "foundryvtt-mindflayer",
+  "version": "1.5.1-development",
```
```suggestion
  "version": "2.0.0-dev",
```

we could change this to 2.0.0-dev(elopment) since this will be a major release

749 replied:
This should be done via the conventional commit automatically

shawly commented on package.json:
```diff
@@ -0,0 +1,39 @@
+  "bugs": {
+    "url": "https://github.com/mindflayer-vtt/foundryvtt-mindflayer/issues"
+  },
+  "homepage": "https://github.com/mindflayer-vtt/foundryvtt-mindflayer#readme",
```
this reminds me that we could create a new issue to setup a gitlab pages project for mindflayer-vtt.github.io and create a nice little portfolio for the project, low prio of course

749 replied:
Created issue #16

shawly commented on src/js/MindFlayer.js:
```diff
@@ -0,0 +1,65 @@
+    }
+  }
+
+  handleCombatUpdate(combat, update) {
```
do we have a function to update keypad LEDs on scene change yet?

749 replied:
The LEDs are updated to the controller in the ControllerManager, if the values have changed.

shawly commented on src/js/dependencies/index.js:
```diff
@@ -0,0 +1,41 @@
+    if (!game.modules.get(dependency)?.active) {
+      if (!game.user.isGM) {
+        ui.notifications.error(
+          `Module '${VTT_MODULE_NAME}' requires the '${dependency}' module. Please install and activate it.`
```
nice

shawly commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,42 @@
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - run: sudo apt update && sudo apt install unzip zip
```
```suggestion
      - run: apt-get update && apt-get install -y unzip zip
```
apt is apparently not recommended for scripting and apt-get should be preferred, also the -y param is probably necessary

shawly commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,42 @@
+name: CI
```
```suggestion
name: CI

env:
  DEBIAN_FRONTEND: noninteractive
```

shawly commented on src/js/module.tmpl.json:
```diff
@@ -0,0 +1,43 @@
+{
+  "name": "mindflayer-token-controller",
+  "title": "Mind Flayer - Token Controller",
+  "description": "Adds a websocket for third party applications to control tokens. Used with the Mind Flayer server.",
```
```suggestion
  "description": "Adds functionality to control tokens through external applications or devices. Mind Flayer server (https://github.com/mindflayer-vtt/mindflayer-server) is required for the module to communicate with external services.",
```

shawly commented on src/js/modules/ControllerManager/Keypad.js:
```diff
@@ -0,0 +1,220 @@
+    return TokenUtil.getTokenFor(player, true);
+  }
+
+  getKeys() {
```
Should we use the same syntax for every get function?
```suggestion
  get keys() {
```

749 replied:
good idea, though that will need to be changed in the other files as well.

shawly commented on src/js/modules/ambilight/index.js:
```diff
@@ -0,0 +1,195 @@
+        image[baseIndex + 1] != 0 ||
+        image[baseIndex + 2] != 0
+      ) {
+        //console.log(x, y, baseIndex, (360 + Math.atan2(direction.y, direction.x)*180/Math.PI) % 360)
```
```suggestion
```
Probably obsolete, right?

shawly commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,42 @@
+      - run: zip -q -r node_modules.zip node_modules
+      - uses: actions/upload-artifact@v2
+        with:
+          name: node_modules.zip
```
```suggestion
          path: node_modules.zip
```

shawly commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,42 @@
+      - uses: actions/checkout@v1
+      - uses: actions/download-artifact@v2
+        with:
+          name: node_modules.zip
```
```suggestion
          path: node_modules.zip
```

shawly committed:
fix: change name to path

shawly committed:
fix: issues with workflow

shawly commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,45 @@
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - run: apt-get update && apt-get install -y unzip zip
```
```suggestion
      - run: sudo apt-get update && sudo apt-get install -y unzip zip
```

shawly commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,45 @@
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - run: apt-get update && apt-get install -y unzip zip
```
```suggestion
      - run: sudo apt-get update && sudo apt-get install -y unzip zip
```

749 commented on src/js/dependencies/index.js:
```diff
@@ -0,0 +1,41 @@
+    }
+    const dependency = moduleJson.dependencies[i].name;
+    if (!game.modules.get(dependency)?.active) {
+      if (!game.user.isGM) {
```
remove the negation
```suggestion
      if (game.user.isGM) {
```

shawly committed:
fix: added sudo to fix permission issues in workflow

shawly committed:
feat: added badges to readme

749 committed:
refactor!: apply suggestions from code review

BREAKING CHANGE: api reworked completely, game.mindflayerctrl is replaced by game.modules.get('mindflayer-token-controller').instance

Co-authored-by: shawly <shawlyde@gmail.com>

shawly committed:
refactor: changed getter syntax for keys
